contracts/rewards/RewardsController.sol

```diff
@@ -264,24 +274,28 @@ contract RewardsController is RewardsDistributor, VersionedInitializable, IRewar
     address user,
     address to
   ) internal returns (address[] memory rewardsList, uint256[] memory claimedAmounts) {
-    _distributeRewards(user, _getUserStake(assets, user));
-
     rewardsList = new address[](_rewardsList.length);
     claimedAmounts = new uint256[](_rewardsList.length);
```

Comment:
Reading length of `_rewardsList` twice, possible to use `rewardsList` length for the second read to not read storage multiple times.